### DISCUSSION TITLE
Show 'too slow' message if no key response in specified time

### DISCUSSION THREAD

**Discussion by SaoirseConnorDesai** [jspsych/jsPsych#2792] [Q&A] [answered]
I'm adapting an experiment someone who was previously in my lab programmed. 

The key difference is that I'm imposing a response deadline so that participants do not respond within X seconds they will a message (new screen) telling them they have been too slow before moving on to the next test trial. 

Someone suggested a loop node to me but that will simply take them back to the original trial and I don't want to do that. 

Any suggestions?

**Comment by ChristopheBossens** [comment, marked as answer]
I think what you need is a [conditional time](https://www.jspsych.org/7.3/overview/timeline/#conditional-timelines). I created a minimal working example that corresponds below. The main idea is that after your trial, you include a node with a conditional function. In this function, the response to the previous trial is evaluated. When this function returns true, it will execute the trials listed in the timeline variable (here, the feedback trial). Let me know if this is what you need!

   ```
 var jsPsych = initJsPsych();
  
    var first_trial = {
		type: jsPsychHtmlKeyboardResponse,
		stimulus: '<p style="font-size:48px; color:green;">BLUE</p>',
		choices: ['r', 'g', 'b'],
		prompt: "<p>Is the ink color (r)ed, (g)reen, or (b)lue?</p>",
		trial_duration: 2000,
	};
	
	var feedback = {
		type: jsPsychHtmlKeyboardResponse,
		stimulus: 'Press a key to continue',
		prompt: "<p>Too slow!</p>"
	}
	
	var evaluate_response_node = {
		timeline: [feedback],
		conditional_function: function(){
			// get the data from the previous trial,
			// and check which key was pressed
			var data = jsPsych.data.get().last(1).values()[0];
			if( data.response == null){
				return true;
			} else {
				return false;
			}
		}
	}
	

	var second_trial = {
		type: jsPsychHtmlKeyboardResponse,
		stimulus: '<p style="font-size:48px; color:blue;">RED</p>',
		choices: ['r', 'g', 'b'],
		prompt: "<p>Is the ink color (r)ed, (g)reen, or (b)lue?</p>",
		trial_duration: 2000
	};
	
	jsPsych.run([first_trial, evaluate_response_node, second_trial]);
```

**Reply by SaoirseConnorDesai** [reply]
Hello! Thank you for this. I ended up changing the task slightly but this will come in handy for the future.
